bazel/grpc_build_system.bzl

```diff
@@ -298,39 +310,58 @@ def grpc_cc_test(name, srcs = [], deps = [], external_deps = [], args = [], data
-        # the vanilla version of the test should run on platforms that only
-        # support a single poller
-        native.cc_test(
-            name = name,
-            testonly = True,
-            tags = (tags + [
-                "no_linux",  # linux supports multiple pollers
-            ]),
-            **args
-        )
-
-        # on linux we run the same test multiple times, once for each poller
-        for poller in POLLERS:
-            native.sh_test(
-                name = name + "@poller=" + poller,
-                data = [name] + data,
-                srcs = [
-                    "//test/core/util:run_with_poller_sh",
-                ],
-                size = size,
-                timeout = timeout,
-                args = [
-                    poller,
-                    "$(location %s)" % name,
-                ] + args["args"],
-                tags = (tags + ["no_windows", "no_mac"]),
-                exec_compatible_with = exec_compatible_with,
-                exec_properties = exec_properties,
-                shard_count = shard_count,
-                flaky = flaky,
-            )
+        # When true, EventEngines will only be exercised against one poller
+        engine_only = False
+        for engine in EVENT_ENGINES:
+            if "no_windows" not in engine["tags"] and "no_mac" not in engine["tags"]:
+                # the vanilla version of the test should run on platforms that only
+                # support a single poller
+                native.cc_test(
+                    name = name,
+                    testonly = True,
+                    tags = (tags + engine["tags"] + [
+                        "no_linux",  # linux supports multiple pollers
+                    ]),
+                    env = {"GRPC_EVENTENGINE_STRATEGY": engine["name"]},
+                    **args
+                )
+
+            if "no_linux" in engine["tags"]:
+                continue
+
+            # on linux we run the same test multiple times, once for each
+            # poller.
+            for poller in POLLERS:
+                native.sh_test(
+                    name = name + "@poller=" + poller + ",engine=" + engine["name"],
+                    data = [name] + data,
+                    srcs = [
+                        "//test/core/util:run_with_poller_sh",
+                    ],
+                    size = size,
+                    timeout = timeout,
+                    args = [
+                        poller,
+                        engine["name"],
+                        "$(location %s)" % name,
+                    ] + args["args"],
+                    tags = (tags + ["no_windows", "no_mac"]),
+                    exec_compatible_with = exec_compatible_with,
+                    exec_properties = exec_properties,
+                    shard_count = shard_count,
+                    flaky = flaky,
+                )
+                # EventEngines only need to be exercised against a single
```

Comment:
Won't this bail after the second iteration of the loop rather than the first?

Also, can you expand this comment to explain why "EventEngines only need to be exercised against a single poller"? It isn't obvious to me from the context.

Reply:
The goal is for every poller to be executed with the default engine, and every other engine to be executed with any one poller arbitrarily. It may be easiest to illustrate by example

```
(dev) ➜  grpc git:(eventengine_end2end_test_setup) bazel query test/cpp/... | grep '//test/cpp/end2end:generic_end2end_test'
...
//test/cpp/end2end:generic_end2end_test@poller=poll,engine=libuv
//test/cpp/end2end:generic_end2end_test@poller=epollex,engine=poll
//test/cpp/end2end:generic_end2end_test@poller=epollex,engine=libuv
//test/cpp/end2end:generic_end2end_test@poller=epoll1,engine=libuv
//test/cpp/end2end:generic_end2end_test
```

Every iomgr poller (`poll`, `epollex`, and `epoll1`) is exercised with the default `libuv` EventEngine, and the `poll` EventEngine is exercised arbitrarily with the `epollex` iomgr engine.

I have however missed duplicating the non-linux targets for each EventEngine, I'll fix that now (the last line should be there once for each engine).